UNO/ClientCommunication/GameSessionControl.java

```diff
@@ -1,5 +1,19 @@
 package ClientCommunication;
 
 public class GameSessionControl {
+    private Client client;
+    private String lobbyId;
+
+    
+    public GameSessionControl(Client client, String lobbyId) {
+        this.client = client;
+        this.lobbyId = lobbyId;
+    }
+
+    
+    public void updateGameState() {
+        Object gameStateUpdate = client.receiveGameState();  // Simulating receiving a game state
+        System.out.println("Game state updated for lobby " + lobbyId + ": " + gameStateUpdate);
+    }
 
```

Comment:
**Recommended to add**

 Fields
    - game session data
    - List of players in the game
    - Index of the current player
 public void addPlayer(ClientHandler player) 

 public void removePlayer(ClientHandler player) 
    
public ClientHandler getCurrentPlayer() 

public void nextTurn() 

public boolean isValidMove(Card card) 
 -Check if the played card is a valid move based on the game rules
 -Example: Check if the card matches the color or value of the top card in the discard pile

public void playCard(ClientHandler player, Card card) 
 - Remove the card from the player's hand
 - Add the card to the discard pile
 -if card type equals "Draw Two, Draw Four", next player draws cards
-Add more conditions for other special card effects

public Card drawCard() 

public void callUno(ClientHandler player) 
- Handle the player calling UNO
-Example: Check if the player has only one card left and apply any penalties if they didn't call UNO in time

public boolean isGameOver() 
-Check if the game is over
-Example: Check if any player has an empty hand

public void broadcastMessage(String message) 
-send players valid messages from server for when they play a bad card, the server will send you this message, you just have to display it

public GameSessionData getGameSessionData() 

private ClientHandler getNextPlayer()